tests/cc/test_usdt_probes.cc

```diff
@@ -293,3 +308,58 @@ TEST_CASE("test listing all USDT probes in Ruby/MRI", "[usdt]") {
     }
   }
 }
+
+TEST_CASE("test probing running Ruby process in namespaces", "[usdt]") {
+  SECTION("in separate mount namespace") {
+    static char _unshare[] = "unshare";
+    const char *const argv[4] = {_unshare, "--mount", "ruby", NULL};
+
+    ChildProcess unshare(argv[0], (char **const)argv);
+    if (!unshare.spawned())
+      return;
+    int ruby_pid = unshare.pid();
+
+    ebpf::BPF bpf;
+    ebpf::USDT u(ruby_pid, "ruby", "gc__mark__begin", "on_event");
+    u.set_probe_matching_kludge(1);  // Also required for overlayfs...
+
+    auto res = bpf.init("int on_event() { return 0; }", {}, {u});
+    REQUIRE(res.msg() == "");
+    REQUIRE(res.code() == 0);
+
+    res = bpf.attach_usdt(u, ruby_pid);
+    REQUIRE(res.code() == 0);
+
+    res = bpf.detach_usdt(u, ruby_pid);
+    REQUIRE(res.code() == 0);
+  }
+
+  SECTION("in separate mount namespace and separate PID namespace") {
+    static char _unshare[] = "unshare";
+    const char *const argv[7] = {_unshare,       "--fork", "--mount", "--pid",
+                                 "--mount-proc", "ruby",   NULL};
+
+    char rubypath[100];
```

Comment:
This is a temporary hack that attempts to find Ruby on Jenkins, but it doesn't return anything... I now suspect these other Ruby tests are somehow skipped by Jenkins?

When I run this test myself with docker on Github actions, such as I have in a draft PR #2714 ruby is located without issue by `unshare` above.